ci/script.sh

```diff
@@ -15,28 +18,23 @@ disable_cross_doctests() {
     fi
 }
 
-run_test_suite() {
-    cargo clean --target $TARGET --verbose
-    cargo build --target $TARGET --verbose
-    cargo test --target $TARGET --verbose
-    cargo build --target $TARGET --verbose --manifest-path grep/Cargo.toml
-    cargo test --target $TARGET --verbose --manifest-path grep/Cargo.toml
-    cargo build --target $TARGET --verbose --manifest-path globset/Cargo.toml
-    cargo test --target $TARGET --verbose --manifest-path globset/Cargo.toml
-    cargo build --target $TARGET --verbose --manifest-path ignore/Cargo.toml
-    cargo test --target $TARGET --verbose --manifest-path ignore/Cargo.toml
-    cargo build --target $TARGET --verbose --manifest-path termcolor/Cargo.toml
-    cargo test --target $TARGET --verbose --manifest-path termcolor/Cargo.toml
-
-    "$( dirname "${0}" )/test_complete.sh"
-
-    # sanity check the file type
-    file target/$TARGET/debug/rg
+run_cargo() {
+    for component in "${components[@]}"; do
+        cargo "${1:?}" --target $TARGET --verbose --manifest-path "${component}/Cargo.toml"
+    done
 }
 
 main() {
     # disable_cross_doctests
-    run_test_suite
+    run_cargo clean
+    run_cargo build
+    if [ "$(architecture)" = "amd64" ] || [ "$(architecture)" = "i386" ]; then
```

Comment:
Does this mean we aren't running tests on arm? Why?

Reply:
In order to run the tests on ARM, you would need to setup `qemu` inside the Travis CI.
This is doable(I've seen it done somewhere) but I don't currently have the knowledge(nor would I like to hold this PR for that) to easily add this feature.

It would definitely add overhead - you would probably need to limit it to run only when preparing a release or something like this.

Reply:
Oh I see, I understand. Thanks for explaining that to me!